tools/devtool

```diff
@@ -411,34 +440,116 @@ validate_kernel_version() {
 #   $2  new version.
 #
 compose_tag_text() {
-    declare prev_ver="$1"
-    declare curr_ver="$2"
-    declare changelog="$FC_ROOT_DIR/CHANGELOG.md"
+    release_text=$(compose_release_text "$1" "$2") || die "Could not compose release description."
+    echo "$release_text" | sed "s/^###\s//g"
+}
 
-    validate_version "$prev_ver"
-    validate_version "$curr_ver"
+# Compose the text for a new release using the information in the changelog,
+# between the two specified releases.
+# Args:
+#   $1  previous version.
+#   $2  new version.
+#
+compose_release_text() {
+    declare curr_ver="$1"
+    declare prev_ver="$2"
+    declare changelog="$FC_ROOT_DIR/CHANGELOG.md"
 
     # Patterns for the sections in the changelog corresponding to the versions.
-    pat_prev="^##\s\[$prev_ver\]"
     pat_curr="^##\s\[$curr_ver\]"
-
+    pat_prev="^##\s\[$prev_ver\]"
     # Extract the section enclosed between the 2 headers and strip off the first
     # 2 and last 2 lines (one is blank and one contains the header `## [A.B.C]`).
     # Then, replace `-` with `*` and remove section headers.
     sed "/$pat_curr/,/$pat_prev/!d" "$changelog" \
       | sed '1,2d;$d' \
-      | sed "s/^-/*/g" \
-      | sed "s/^###\s//g"
+      | sed "s/^-/*/g"
+}
+
+get_prev_release() {
+    declare version=$1
+    declare pat_release="^## \[([0-9]+\.){2}[0-9]+\]"
+    declare changelog="$FC_ROOT_DIR/CHANGELOG.md"
+
+    grep -q "\[$version\]" "$changelog"
+    ok_or_die "No changelog entry for release $version can be found. Make sure you have run ./devtool prepare_release."
+    # We work with the assumption that the changelog has already been updated
+    # and contains a header (and a corresponding section) for the new release.
+
+    # Step 1: Get all release numbers.
+    all_releases=($(grep -E "$pat_release" "$changelog"))
+    # Step 2: Trim out headers (`##`).
+    all_releases=(${all_releases[@]//##*})
+
+    # Step 3: Walk the array until we come across the desired release number,
+    # then pick up the next one. Since the latest releases are at the top of the
+    # changelog, the next one in line will be the previous one chronologically.
+    # The array now contains all the release numbers in the changelog, enclosed
+    # in square brackets.
+    for release in "${all_releases[@]}"; do
+        if [ -n "$found" ]; then
+            # Trim out square brackets.
+            prev_version=$(echo "$release" | awk -F"[][]" "{print \$2}")
+            break
+        elif [ "$release" == "[$version]" ]; then
+            found=1
+        fi
+    done
+
+    if [ -z "$prev_version" ]; then
+        die "Could not find a previous release for v$version."
+    fi
+
+    echo $prev_version
+}
+
+# Auxiliary function for creating the JSON for calling into
+# GitHub's API for posting a release.
+generate_release_post_data() {
+  cat <<EOF
+{
+  "tag_name": "v$1",
+  "target_commitish": "$2",
+  "name": "Firecracker v$1",
+  "body": "$4",
+  "draft": $draft,
```

Comment:
I'd rather have the `$draft` flag as a parameter

Reply:
The reason I hardcoded was to not run the risk to post a release that is not a draft. A release that is not a draft will be visible for the entire community. Let me know if you think we should still make it a parameter.